### PR TITLE
Remove Sponsor

### PR DESCRIPTION
**Pull request by nflearl**
**JIRA Tickets:**

[AB2D-42](https://jira.cms.gov/browse/AB2D-2715) - Remove Sponsor

***Related Tickets***
 
### What Does This PR Do?

Removes the Sponsor domain object from the system.  Now, a User maps directly to a contract and all jobs are directly tied to a single contract.

### What Should Reviewers Watch For?

There are a lot of tricky assumptions built into the test suite.  There is also assumptions built in about being able to select which contract for a job.  Now it is hardwired.  Admin running a job on behalf of a User still works since effectively, the target User is used for job submission. 
### Database Changes

Applied in previous PRs and in this one:
Drop NOT NULL constraints for sponsor references and for the new Contract reference.  Added a FK constraint for contract reference in the User table.

### Security Implications
It is more secure since a user can now only run a job with the associated contract.  Specifying a contract in the API simply results in a check against the configured contract.  There is now no way to run a job against another contract.

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [x] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [x] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java [hunk 1]
```diff
@@ -108,19 +107,21 @@ public Job process(final String jobUuid) {
...
      * @param progressTracker - the progress tracker which indicates how far the job is along
-     * @throws ExecutionException when there is an issue with searching
+     * @throws ExecutionException   when there is an issue with searching
      * @throws InterruptedException - when the search is interrupted
      */
-    void processContract(Contract contract, Job job, int month, Path outputDirPath, ProgressTracker progressTracker) throws ExecutionException, InterruptedException {
+    void processContract(Job job, int month, Path outputDirPath, ProgressTracker progressTracker)
+            throws ExecutionException, InterruptedException {
+        Contract contract = job.getContract();
+        assert contract != null;
```
Comment (wnyffenegger) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java:120]: Missed the assert

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java [hunk 2]
```diff
@@ -146,7 +147,10 @@ void processContract(Contract contract, Job job, int month, Path outputDirPath,
                 progressTracker.getFailureCount()));
     }
 
-    void processContractBenes(Job job, Contract contract, int month, ProgressTracker progressTracker) throws ExecutionException, InterruptedException {
+    void processContractBenes(Job job, int month, ProgressTracker progressTracker)
+            throws ExecutionException, InterruptedException {
+        Contract contract = job.getContract();
+        assert contract != null;
```
Comment (wnyffenegger) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java:153]: Another assert in here

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java (hunk 1 above)
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java:120]: Asserts are intentional.  With the new assumption that a User is bound to a single Contract, results in a Job being bound to a single contract.

The contract reference must exist.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java (hunk 2 above)
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java:153]: Asserts are intentional.  With the new assumption that a User is bound to a single Contract, results in a Job being bound to a single contract.

The contract reference must exist.

**Review by wnyffenegger — APPROVED** [review]

**Review by lhanekam — APPROVED** [review]
Database change looks good

**Review by lailasharshar — APPROVED** [review]

**Comment by nflearl** [comment]
Remove all things Sponsor from the code base.  The table is left in place for now as an emergency rollback.

**Pull request merged**